### ISSUE TITLE
Class Plan for Wednesday, Jan 24

### ISSUE THREAD

**Issue by steven-king** [steven-king/mj583#5]
To start class:
Launch and login to Docker App.

Pull the image using

``` docker pull uncdataviz/class-03:start```

**Conceptual Lecture**
[Understanding Scrapers
Ethics of Scraping](https://github.com/steven-king/mj583/wiki/Concepts-and-Ethics-of-Web-Scraping)

**Technical Lecture**
[Writing a Scraper using Scrapy](https://github.com/steven-king/mj583/wiki/Using-Scrapy-in-Python)

To Start the container using
```
docker run -p 9000:9000 -d -t uncdataviz/class-03:start
```

**Comment by steven-king** [comment]
I have added my iPython notebook that has lots of comments. Note that I use multiple cell to show how it builds and changes but you don't have to duplicate all that work.  This also covers what we are going to be doing on Monday.

**Comment by steven-king** [comment]
Assignment:
scrape a roster from goheels.com and store the player name and href as a key value par list. 

Do this in an iPython notebook. Remember to use docker commit to save the state and also export the notebook before killing the process in the terminal.

Just bring to class. You do not have to turn it in.

**Comment by jtizon001** [comment]
I am having trouble making a connection to goheels.com. I keep getting 404 error responses yet if i curl it through terminal it works fine. I am wondering if there is an issue with how the server is handling requests. Anyone else having this problem?

**Comment by aryaswanie** [comment]
Yes I’ve been having the same problem! All other urls work fine but goheels.com never works.

Arya

**Comment by steven-king** [comment]
I will look into this very late tonight can get back to the class.

Has anyone been successful?

Sent from my iPhone

On Jan 27, 2018, at 4:45 PM, aryaswanie <notifications@github.com<mailto:notifications@github.com>> wrote:

Yes I’ve been having the same problem! All other urls work fine but goheels.com<http://goheels.com> never works.

Arya

**Comment by elisabeth-parker** [comment]
I'm having the same problem, 404 error from goheels.com but success from other sites.

**Comment by steven-king** [comment]
CHANGE of PLAN: It looks like our agent is being blocked. You do not need to do the assignment. Just study the Cody from Wednesday and make sure you understand what is going on in each cell.

Sorry,
Steven

Sent from my iPhone

On Jan 28, 2018, at 10:09 AM, elisabeth-parker <notifications@github.com<mailto:notifications@github.com>> wrote:


I'm having the same problem, 404 error from goheels.com<http://goheels.com> but success from other sites.
